app/views/studies/_shared_sync_functions.js.erb

```diff
@@ -1,9 +1,81 @@
 // shared JS functions for sync forms
 
-$('#study-file-<%= study_file.id %>').on('change', '.file-type', function() {
+// Used for sync CSFV
+if (typeof window.SCP.readOnlyToken === 'undefined') {
+  window.SCP.readOnlyToken = '<%= get_read_access_token(@study, current_user) %>'
+}
+
+/**
+* Transform validation results to HTML that can be shown to user.
+*
+* This function is a temporary backport of the issue messaging from
+* FileUploadControl.js.  It should be replaced when sync code migrates to
+* React.
+*/
```

Comment:
BLOCKER: Instead of duplicating all this, let's extract lines 151-164 of FileUploadControl into a new component `<ValidationMessages fileValidation={fileValidation}/>`. Then make a helper method `window.SCP.renderValidationMessages(target, fileValidation)` (similar to the `UploadWizard.renderUploadWizard` helper method).  You're right this may be temporary, but that render helper pattern is an important one for us to master as we're steadily migrating functionality over to React

Reply:
Sorry for the confusion due to the vague ticket.  In the ticket, 'legacy UI' just meant that we were working inside existing legacy UI, instead of upgrading the whole thing to react,  and so would be reliant on legacy ways of, e.g. listening for menu changes.  We don't want to create *new* legacy UI structures that are duplicates of React components unless we absolutely have to, and here we don't have to.  Especially since this validation messaging is going to be under active development over the next months, tracking it in two places will make that work more difficult.

And I can't emphasize enough how much the render helper pattern will be important to migrating more of SCP to react/API patterns in an efficient way.  Right now, I'm assuming you did it the way you did because it was faster and easier for you.  We need to get to the point where it's faster and easier for you to say 'oh, I'll just reuse our existing component for that'.  I'm happy to help out any way I can -- grabbing an existing react component and rendering on a legacy UI template shouldn't take more than 10 minutes

Reply:
No worries, its difficult assess how much debt tech is reasonable for a functional MVP, especially when the ticket can be reasonably read a few different ways on the matter!  

I do like the pattern you introduced, but I do think most engineers with experience in both jQuery and React would find it faster and easier to do basic templating in jQuery when working in code that's in jQuery.

Per our chat I went ahead and implemented that new React component.  I also replaced up the redundant vestigial component and updated its test to apply to the new component.